google-cloud-datastore/OVERVIEW.md

```diff
@@ -0,0 +1,579 @@
+# Google Cloud Datastore
+
+Google Cloud Datastore is a fully managed, schemaless database for storing
+non-relational data. You should feel at home if you are familiar with relational
+databases, but there are some key differences to be aware of to make the most of
+using Datastore.
+
+The goal of google-cloud is to provide an API that is comfortable to Rubyists.
+Your authentication credentials are detected automatically in Google Cloud
+Platform environments such as Google Compute Engine, Google App Engine and
+Google Kubernetes Engine. In other environments you can configure authentication
+easily, either directly in your code or via environment variables. Read more
+about the options for connecting in the {file:AUTHENTICATION.md Authentication
+Guide}.
+
+```ruby
+require "google/cloud/datastore"
+
+datastore = Google::Cloud::Datastore.new(
+  project_id: "my-todo-project",
+  credentials: "/path/to/keyfile.json"
+)
+
+task = datastore.find "Task", "sampleTask"
+task["priority"] = 5
+datastore.save task
+```
+
+To learn more about Datastore, read the
+[Google Cloud Datastore Concepts Overview
+](https://cloud.google.com/datastore/docs/concepts/overview).
+
+## Retrieving records
+
+Records, called "entities" in Datastore, are retrieved by using a key. The key
+is more than a numeric identifier, it is a complex data structure that can be
+used to model relationships. The simplest key has a string `kind` value and
+either a numeric `id` value or a string `name` value. A single record can be
+retrieved by calling {Google::Cloud::Datastore::Dataset#find} and passing the
+parts of the key:
+
+```ruby
+require "google/cloud/datastore"
+
+datastore = Google::Cloud::Datastore.new
+
+task = datastore.find "Task", "sampleTask"
+```
+
+Optionally, {Google::Cloud::Datastore::Dataset#find} can be given a key object:
```

Comment:
All of these links to code constructs will definitely not work when this file is viewed in github's markdown rendering, but what about in the YARD-generated HTML?

Reply:
The links will work in the YARD-generated HTML, which is a big part of why I'm proposing this.